backend/src/main/java/com/mcmasterbaja/FileUploadResource.java

```diff
@@ -58,17 +61,11 @@ public void uploadFile(
 
         try (InputStream input = fileData) { // try-with-resources, look it up if you don't know it
           BinaryToCSV.bytesToCSV(fileData.readAllBytes(), outputDir, fileName, true);
-        } catch (IOException e) { // UnsatisfiedLinkError, IOException
-          throw new StorageException("Failed to read bytes from: " + fileName, e);
         }
         break;
 
       default:
-        try {
-          fileData.close();
-        } catch (IOException e) {
-          throw new StorageException("Failed to close fileData", e);
-        }
+        fileData.close();
```

Comment:
This is def on me, but I think introducing a method within the storageService to handle this binToCsv stuff might be a better idea. Any business logic should be extracted out of the Resources